### DISCUSSION TITLE
How to start a docker based dogecoin full node mainnet

### DISCUSSION THREAD

**Discussion by Palak3196** [dogecoin/dogecoin#3539] [Help needed]
Hi Team,

Looking for resources that I can use (compose file or dockerfile to build an image from) to run a Dogecoin mainnet full node (having data from genesis). Space and CPU/RAM are not an issue for me.

I hope this is achievable, though I do not see a recently upgraded dockerfile in the repo for Release 1.14.7. Please guide.

Thanks,
Palak

**Comment by GreatApe42069** [comment]
https://github.com/GreatApe42069/NodeRunnerDocker.git

**Comment by Palak3196** [comment]
@GreatApe42069 Thanks for the link. Is this an official setup or are you a developer from Dogecoin team itself?

**Comment by Palak3196** [comment]
@GreatApe42069, I tried this. Successfully built the image and started the container with docker-compose.
But getting this error: 

```
dogecoin_node  | error: couldn't connect to server: unknown (code -1)
dogecoin_node  | (make sure server is running and you are connecting to the correct RPC port)
```

**Comment by Palak3196** [comment]
Hi Team, any answers for me?

**Reply by daank-c** [reply]
Hey Palak,

First, I just want to emphasize that in the world of permissionless/consensus-driven software like Dogecoin, there are no "official" implementations, repositories, or representatives. This is not just an ethical stance; it’s a fundamental aspect of the design that ensures power and control remain distributed, helping prevent single points of failure and some degree of resistance to corporate influence.

In regards to your question, I don't have the exact resource you're looking for, but I can point you towards a couple of things related to containerization and Dogecoin Core:

Community Repository: There's a repository called 'docker' where people share their Docker images for Dogecoin Core. This repository is **not stable**, and as such, any contributions you could make would be incredibly valuable. If you find something that works or develop something new, sharing it [here](https://github.com/dogecoin/docker) would be fantastic.

Containerized Implementations: I'm aware of a few containerized applications that utilize Dogecoin Core containers, notably [BTCPayServer](https://github.com/btcpayserver/dockerfile-deps) and [Thorchain's Node Launcher](https://gitlab.com/thorchain/devops/node-launcher/-/tree/master). Both of these use Dogecoin Core 1.14.7, though I believe Thorchain's Node Launcher is designed for Kubernetes environments rather than Docker.

These repositories might serve as helpful references for writing your own Dockerfile or modifying an existing one to suit your needs.

Best of luck with your setup!
